### PR TITLE
remove deprecated MessageType

### PR DESCRIPTION
**Pull request by rokkiter**
**Please provide a description of this PR:**

proto.MessageType is deprecated，use protoregistry.GlobalTypes.FindMessageByName replace it

### CODE REVIEW

**Comment by hanxiaop** [comment]
@rokkiter `ok-to-test` won't work if you are not a member of Istio, I encourage you to apply for the membership.

File: istioctl/pkg/util/configdump/wrapper.go [hunk 1]
```diff
@@ -37,13 +38,12 @@ func (m *nonstrictResolver) Resolve(typeURL string) (legacyproto.Message, error)
...
-	// nolint: staticcheck
-	mt := legacyproto.MessageType(mname)
-	if mt == nil {
+	mt, err := protoregistry.GlobalTypes.FindMessageByName(protoreflect.FullName(mname))
+	if err != nil {
 		// istioctl should keep going if it encounters new Envoy versions; ignore unknown types
 		return &exprpb.Type{TypeKind: &exprpb.Type_Dyn{Dyn: &emptypb.Empty{}}}, nil
 	}
-	return reflect.New(mt.Elem()).Interface().(legacyproto.Message), nil
+	return legacyproto.MessageV1(mt.New().Interface()), nil
```
Comment (hanxiaop) [inline istioctl/pkg/util/configdump/wrapper.go:46]: Have you tried this? I've tested this and gives me a panic:
```
istioctl x authz check httpbin-istio-waypoint-56cd46b558-ftzjl.ambient
panic: message *impl.MessageInfo is neither a v1 or v2 Message

goroutine 1 [running]:
google.golang.org/protobuf/internal/impl.Export.ProtoMessageV1Of({}, {0x103651540?, 0x140001cb248})
        google.golang.org/protobuf@v1.30.0/internal/impl/api_export.go:106 +0x128
github.com/golang/protobuf/proto.MessageV1(...)
        github.com/golang/protobuf@v1.5.3/proto/proto.go:55
istio.io/istio/istioctl/pkg/util/configdump.(*nonstrictResolver).Resolve(0x14000110780?, {0x1400086a3c0, 0x36})
        istio.io/istio/istioctl/pkg/util/configdump/wrapper.go:49 +0x90
github.com/golang/protobuf/jsonpb.(*Unmarshaler).unmarshalMessage(0x14000890768, {0x10373b2a0, 0x14000110780}, {0x14001260000, 0x1184e, 0x12000})
        github.com/golang/protobuf@v1.5.3/jsonpb/decode.go:165 +0x290
github.com/golang/protobuf/jsonpb.(*Unmarshaler).unmarshalSingularValue(0x10342b5e0?, {{}, 0x10360a180?, 0x14000110780?, 0x102eba0a0?}, {0x14001260000, 0x1184e, 0x12000}, {0x10374c528, 0x14000b10ff0})
        github.com/golang/protobuf@v1.5.3/jsonpb/decode.go:499 +0x3cc
github.com/golang/protobuf/jsonpb.(*Unmarshaler).unmarshalValue(0x140009ec040?, {{}, 0x10342b5e0?, 0x1400085e630?, 0x7?}, {0x14001240000, 0x1f473, 0x20000}, {0x10374c528, 0x14000b10ff0?})
        github.com/golang/protobuf@v1.5.3/jsonpb/decode.go:418 +0x520
github.com/golang/protobuf/jsonpb.(*Unmarshaler).unmarshalMessage(0x14000890768, {0x10373b2a0, 0x140009ec040}, {0x14001220000, 0x1f483, 0x20000})
        github.com/golang/protobuf@v1.5.3/jsonpb/decode.go:340 +0x1774
github.com/golang/protobuf/jsonpb.(*Unmarshaler).UnmarshalNext(0x10341c0e0?, 0x140009ec040?, {0x12dd90170, 0x140009ec040})
        github.com/golang/protobuf@v1.5.3/jsonpb/decode.go:127 +0x154
github.com/golang/protobuf/jsonpb.(*Unmarshaler).Unmarshal(...)
        github.com/golang/protobuf@v1.5.3/jsonpb/decode.go:68
istio.io/istio/pkg/util/protomarshal.UnmarshalAllowUnknownWithAnyResolver({0x1036f07e0?, 0x1054f6ca0?}, {0x140011aa000?, 0x1f484?, 0x20000?}, {0x1036db520?, 0x140009ec040?})
        istio.io/istio/pkg/util/protomarshal/protomarshal.go:59 +0x148
istio.io/istio/istioctl/pkg/util/configdump.(*Wrapper).UnmarshalJSON(0x14000490068, {0x140011aa000, 0x1f484, 0x20000})
        istio.io/istio/istioctl/pkg/util/configdump/wrapper.go:66 +0x64
istio.io/istio/istioctl/cmd.getConfigDumpFromPod({0x16f78f707, 0x27}, {0x16f78f72f, 0x7})
        istio.io/istio/istioctl/cmd/authz.go:138 +0x27c
istio.io/istio/istioctl/cmd.glob..func2(0x1054894a0?, {0x14000c33510, 0x1?, 0x1?})
        istio.io/istio/istioctl/cmd/authz.go:78 +0x150
github.com/spf13/cobra.(*Command).execute(0x1054894a0, {0x14000c334e0, 0x1, 0x1})
        github.com/spf13/cobra@v1.7.0/command.go:940 +0x5c4
github.com/spf13/cobra.(*Command).ExecuteC(0x14000ed4000)
        github.com/spf13/cobra@v1.7.0/command.go:1068 +0x340
github.com/spf13/cobra.(*Command).Execute(0x1400006e0b0?)
        github.com/spf13/cobra@v1.7.0/command.go:992 +0x1c
main.main()
        istio.io/istio/istioctl/cmd/istioctl/main.go:39 +0xd0
```

Comment (hanxiaop) [reply istioctl/pkg/util/configdump/wrapper.go:46]: Looks like we need to keep the original messageType

Comment (rokkiter) [reply istioctl/pkg/util/configdump/wrapper.go:46]: please pull and try again, i have tested in local and it without panic

Comment (hanxiaop) [reply istioctl/pkg/util/configdump/wrapper.go:46]: Now it works

**Review by hanxiaop — APPROVED** [review]
lgtm, thanks

**Pull request merged**